content/advent-2016/testing-distributed-systems-in-go.md

```diff
@@ -0,0 +1,185 @@
++++
+author = ["Gyu-Ho Lee"]
+date = "2016-12-27T08:00:00+00:00"
+title = "Testing distributed systems in Go"
+series = ["Advent 2016"]
++++
+
+
+### What is etcd
+
+[etcd](https://github.com/coreos/etcd) is key-value store for the most critical data of distributed systems. Some use cases are: Applications running on [cotainer linux by CoreOS](https://coreos.com/why) gets automatic Linux kernel updates. CoreOS uses etcd to store semaphore values to make sure only subset of cluster are rebooting at any given time. [Kubernetes](http://kubernetes.io/docs/whatisk8s) uses etcd to store cluster states for service discovery and cluster management, and it uses watch API to monitor critical configuration changes. Consistency is the key to ensure that services correctly schedule and operate.
```

Comment:
`Consistency is the key to ensure that services correctly schedule and operate.` => 
`Consistency is the key to ensure those services schedule and operate correctly.`